src/components/auth/RegisterForm.js

```diff
@@ -5,65 +5,129 @@ import { connect } from 'react-redux';
 import Form from 'react-validation/build/form';
 import Input from 'react-validation/build/input';
 import Button from 'react-validation/build/button';
-import type { UserParams } from '../../models/User';
 import { register } from '../../state/auth/auth.actionCreators';
-import './RegisterForm.scss';
+import './Forms.scss';
 
 type Props = {
   dispatch: Function,
 };
 
 type State = {
-  user: UserParams,
+  email: string,
+  username: string,
+  password: string,
+  passwordConfirm: string,
+  errors: Object,
 };
 
 class RegisterForm extends Component<Props, State> {
   constructor(props) {
     super(props);
 
     this.state = {
-      user: {
-        email: '',
-        username: '',
-        password: '',
-      },
+      email: '',
+      username: '',
+      password: '',
+      passwordConfirm: '',
+      errors: {},
     };
   }
 
-  handleChange = (event) => {
-    const { name, value } = event.target;
-    const { user } = this.state;
+  handleChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.resetErrors();
+
+    const {
+      email,
+      username,
+      password,
+      passwordConfirm,
+    } = this.state;
+    const { dispatch } = this.props;
+
+    if (email && username && password && passwordConfirm) {
+      if (password !== passwordConfirm) {
+        this.handleError('mismatchedPassword', true);
+        return false;
+      }
+
+      if (!this.isValidEmail(email)) {
+        this.handleError('invalidEmail', true);
+        return false;
+      }
+
+      const user = { email, username, password };
+
+      dispatch(register(user));
+    } else {
+      this.handleError('missingInput', true);
+    }
+    return true;
+  }
+
+  handleError = (name, value) => {
     this.setState({
-      user: {
-        ...user,
+      errors: {
         [name]: value,
       },
     });
   }
 
-  handleSubmit = (event) => {
-    event.preventDefault();
+  resetErrors = () => {
+    this.setState({
+      errors: {},
+    });
+  }
 
-    const { user } = this.state;
-    const { dispatch } = this.props;
-    if (user.username && user.email && user.password) {
-      dispatch(register(user));
+  isValidEmail = (email) => {
```

Comment:
This should not be in this class in my opinion.
Consider using a package for that.